library/aes.c

```diff
@@ -831,6 +835,9 @@ int mbedtls_aes_setkey_enc( mbedtls_aes_context *ctx, const unsigned char *key,
 #endif
     ) )
     {
+#if defined(MBEDTLS_VALIDATE_AES_KEYS_INTEGRITY)
+        ctx->crc = mbedtls_crc_update( 0, ctx->rk, keybits >> 3 );
+#endif
```

Comment:
@AndrzejKurek , it seems we don't need "ret= 0" in line 828, and "return ret", we can return 0

Reply:
This is a part of FI countermeasures, so that one cannot skip to `return 0` from the beginning of the function, but has to step to the one place that changes it to 0. Would you like me to change it?

Reply:
I would insert CRC check to be in "if" of flow_counter check, so only if flow counter is fine zero will be return. In this case it cannot jump to end and just return zero

Reply:
It is "if" of flow counter, so it can not return 0 if counter is wrong. It seems it is fine to remove ret from this function

Reply:
Done.